modules/sfp_spur.py

```diff
@@ -0,0 +1,222 @@
+            self.sf.error("Unable to fetch data from spur.us", False)
+            return None
+        
+        content = res.get('content')
+
+        return content
+
+    def handleEvent(self, event):
+        
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+        
+        if self.errorState:
+            return None
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Always check if the API key is set and complain if it isn't, then set
+        # self.errorState to avoid this being a continual complaint during the scan.
+        if self.opts['api_key'] == "":
+            self.sf.error("You enabled sfp_spur but did not set an API key!", False)
+            self.errorState = True
+            return None
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+        else:
+            self.results[eventData] = True
+
+
+        if eventName == 'NETBLOCK_OWNER':
+            if not self.opts['netblocklookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxnetblock']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxnetblock']))
+                    return None
+        
+        if eventName == 'NETBLOCK_MEMBER':
+            if not self.opts['subnetlookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxsubnet']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxsubnet']))
+                    return None
+                    
+        qrylist = list()
+        if eventName.startswith("NETBLOCK_"):
+            for ipaddr in IPNetwork(eventData):
+                qrylist.append(str(ipaddr))
+                self.results[str(ipaddr)] = True
+        else:
+            # If user has enabled affiliate checking
+            if eventName == "AFFILIATE_IPADDR" and not self.opts['checkaffiliates']:
+                return None
+            qrylist.append(eventData)
+        
+        for addr in qrylist:
+
+            if self.checkForStop():
+                return None
+            
+            data = json.loads(self.queryIPAddress(addr))
+
+            if data is None:
+                continue
+            
+            if eventName.startswith("NETBLOCK_"):
+                ipEvt = SpiderFootEvent("IP_ADDRESS", addr, self.__name__, event)
+                self.notifyListeners(evt)
+
+            evt = SpiderFootEvent("RAW_RIR_DATA", str(data), self.__name__, event)
```

Comment:
Should probably use `ipEvt` instead of `event`, since it was the IP address you looked up, not the netblock.